### DISCUSSION TITLE
Interesting behaviour I can't understand

### DISCUSSION THREAD

**Discussion by RazvanManolache** [cssbattle/forum#146] [Q&A] [answered]
So I've been revisiting some older targets, see if I learned anything (a little, nothing amazing) and I don't understand something.

for target 14 i've been trying unicode. It's obviously a wrong path, but you know, learning. I've found out that unicode triangles for up and down do not line up. And while trying to line them up I see that margin doesn't do squat (see the 9999999)

`▼<a>▲<style>*{background:#F2F2B6;color:#FF6D00;transform:scale(3.9,3.7);text-shadow:1.4px 0#FD4602;margin:69.3 93.9}a{background:#0000;color:#FD4602;margin:-9999999-8.9;line-height:22px;text-shadow:1.4px 0#FF6D00}`

Anybody got any idea why?

I've also tried for target 4 a unicode version, circle and square are not the same width. Was unicode drawn by a child? Sure, how we use it is somehow insane, but for typography and stuff... this is crap.

**Comment by Beowolve** [comment, marked as answer]
The `<a>` tag is by default an inline element, and only the x direction can be modified by default. It's sometimes useful because u can then just write `margin:<x>` and only modify the x-axis position.
You could use position:fixed and margin will work quite nice then. Also `float:left` will work, but it is better to switch to a different tag that has better default values for your usage, so you can skip that code all together. There is another single character tag that works quite good, you will figure it out ;).

Here are some other general tips to optimize your code:
* You can dump the "#" char for color
* transform:scale and margin can be combined with transform:matrix
* `background:#0000` can be replaced with `background:0` (In this case, set the background color on body only, and you don't need that)

**Comment by RazvanManolache** [comment]
I hate matrix :)))

I see what you mean, still haven't managed to break 99.9% with this method. Then I'll see how to simplify it. 131 more points and you're going down :)

**Reply by Beowolve** [reply]
> I hate matrix :)))
> 
> I see what you mean, still haven't managed to break 99.9% with this method. Then I'll see how to simplify it. 131 more points and you're going down :)

There are many matrix generators available, I use this one: 
[https://angrytools.com/css-generator/transform/](https://angrytools.com/css-generator/transform/)

For target 14 the 99.9% problem happend to me a lot too, try shifting everything to the left by 1 pixel -> 💯 

![image](https://user-images.githubusercontent.com/32326640/139067132-6ade92ca-e1b2-4966-b97f-ef142914082c.png)

**Reply by SelenIT** [reply]
Another "lifehack" for generating matrix transforms is writing all needed transforms in longhand form, then opening the "computed style" tab in the devtools and copying the value of the `transform` property from there. The computed value of `transform` is always the resulting `matrix` (which is the product of matrices corresponding to all individual transforms applied to the element). You will only need to manually fix the rounding errors (things like `6e-24` instead of `0`) and remove non-significant decimals (2 or 3 digits after the dot should be enough usually) and whitespaces.

**Reply by RazvanManolache** [reply]
Damn, I have matrix on none of my solutions. Maybe I should consider it more. 

This is actually a pretty cool hack.

**Reply by gunnarbittersmann** [reply]
> I hate matrix :)))

Why are you using characters that need different scaling in _x_ and _y_ then?

**Reply by RazvanManolache** [reply]
is there another triangle i don't know about? :))

**Reply by gunnarbittersmann** [reply]
[unicode.org](https://home.unicode.org/) has a search field on the page (even though at an uncommon place).

**Reply by gunnarbittersmann** [reply]
But thanks for making me revisit this target. It paid off. ☺️

**Reply by Beowolve** [reply]
congrats @gunnarbittersmann 👍
